jaeger-core/src/main/java/com/uber/jaeger/Configuration.java

```diff
@@ -385,6 +419,201 @@ public Integer getMaxQueueSize() {
     }
   }
 
+  /**
+   * Holds the configuration related to the sender. A sender can be a {@link HttpSender} or {@link UdpSender}
+   *
+   */
+  public static class SenderConfiguration {
+    /**
+     * A custom sender set by our consumers. If set, nothing else has effect. Optional.
+     */
+    private Sender sender;
+
+    /**
+     * The Agent Host. Has no effect if the sender is set. Optional.
+     */
+    private String agentHost;
+
+    /**
+     * The Agent Port. Has no effect if the sender is set. Optional.
+     */
+    private Integer agentPort;
+
+    /**
+     * The endpoint, like https://jaeger-collector:14268/api/traces
+     */
+    private String endpoint;
+
+    /**
+     * The Auth Token to be added as "Bearer" on Authorization headers for requests sent to the endpoint
+     */
+    private String authToken;
+
+    /**
+     * The Basic Auth username to be added on Authorization headers for requests sent to the endpoint
+     */
+    private String authUsername;
+
+    /**
+     * The Basic Auth password to be added on Authorization headers for requests sent to the endpoint
+     */
+    private String authPassword;
+
+    /**
+     * Creates a new configuration based on the given sender.
+     * @param sender the sender to use
+     */
+    public SenderConfiguration(Sender sender) {
+      this.sender = sender;
+    }
+
+    /**
+     * Creates a new Sender configuration sending data to an agent via UDP
+     * @param agentHost the agent's hostname
+     * @param agentPort the agent's port
+     */
+    public SenderConfiguration(String agentHost, Integer agentPort) {
```

Comment:
considering that we're extending the API surface, maybe it's a good time to take a different approach to these classes? Having overloaded constructors with very similar arguments is far from ideal. I would rather define a builder with explicit methods like `builder.withUDP(String agentHost, Integer agentPort)`, `withHTTP(String endpoint)`, withBasicAuth, withAuthToken, etc.

Reply:
Agree with having a builder approach, but I'd stick with the way things are now, for consistency's sake, and then add a builder later.

Reply:
Well the problem it's that once we release this version it becomes public API and a  maintenance headache. Builder is a cleaner API.

Reply:
Done, the latest version of this PR has a builder pattern for the `SenderConfiguration`.